RecoLocalTracker/SiPixelClusterizer/interface/SiPixelFedCablingMapGPUWrapper.h

```diff
@@ -33,21 +33,15 @@ class SiPixelFedCablingMapGPUWrapper {
 
 private:
   const SiPixelFedCablingMap *cablingMap_;
-  std::vector<unsigned int,  CUDAHostAllocator<unsigned int>>  fedMap;
-  std::vector<unsigned int,  CUDAHostAllocator<unsigned int>>  linkMap;
-  std::vector<unsigned int,  CUDAHostAllocator<unsigned int>>  rocMap;
-  std::vector<unsigned int,  CUDAHostAllocator<unsigned int>>  RawId;
-  std::vector<unsigned int,  CUDAHostAllocator<unsigned int>>  rocInDet;
-  std::vector<unsigned int,  CUDAHostAllocator<unsigned int>>  moduleId;
-  std::vector<unsigned char, CUDAHostAllocator<unsigned char>> badRocs;
   std::vector<unsigned char, CUDAHostAllocator<unsigned char>> modToUnpDefault;
   unsigned int size;
   bool hasQuality_;
 
+  SiPixelFedCablingMapGPU *cablingMapHost = nullptr; // pointer to struct in CPU
+
   struct GPUData {
     ~GPUData();
-    SiPixelFedCablingMapGPU *cablingMapHost = nullptr;   // internal pointers are to GPU, struct itself is on CPU
-    SiPixelFedCablingMapGPU *cablingMapDevice = nullptr; // same internal pointers as above, struct itself is on GPU
+    SiPixelFedCablingMapGPU *cablingMapDevice = nullptr; // pointer to struct in GPU
```

Comment:
last comment not relevant to this PR, but rather to `CUDAESProduct` in general :-)

The pattern seems to be:
  - start with a class/struct for the actual data on th GPU
```c++
  Payload *cablingMapHost = nullptr; // pointer to struct in CPU
```

   - define a wrapper
```c++
  struct PayloadWrapper {
    ~PayloadWrapper();
    Payload *payload = nullptr; // pointer to struct in GPU
  };
```

  - add a CUDAESProduct data mamber:
```c++
  CUDAESProduct<PayloadWrapper> payload_;
```

  - produce it for the gpu like this
```c++
Payload const* getGPUProductAsync(cuda::stream_t<>& cudaStream) const {
  const auto& data = payload_.dataForCurrentDeviceAsync(cudaStream, [this](PayloadWrapper& data, cuda::stream_t<>& stream) {
      // allocate
      cudaCheck(cudaMalloc(&data.payload, sizeof(Payload)));

      // transfer
      cudaCheck(cudaMemcpyAsync(data.payload, this->cablingMapHost, sizeof(Payload), cudaMemcpyDefault, stream.id()));
  });
  return data.payload;
}
```

Would it make sense to encapsulate more of the common part into `CUDAESProduct` ?

And/or to drop the PayloadWrapper in favour of a unique_ptr, possibly with a custom destructor ?

Reply:
I'm all for encapsulating patterns, but I need to think this for a while. I made an issue #336 of it to remind.